Add permissions to the Codecov API endpoints

michelletran-sentry opened the pull request:
Adding some permissions to ensure that Codecov Endpoints checks the following:

1. User is authenticated
2. User has permissions on the org
3. And the integration is associated with the org

Tagging the @getsentry/ecosystem team for review because we're inheriting the `OrganizationEndpoint` to do permissions check, and the integrations check.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

michelletran-sentry committed:
Add permissions to the Codecov API endpoints

ajay-sentry commented on src/sentry/codecov/base.py:
```diff
@@ -2,17 +2,33 @@
-        # TODO: in case we need to modify args, do it here
+
+        owner = parsed_kwargs.get("owner")
+        organization = parsed_kwargs.get("organization")
```
is the organization automagically loaded from the organization_id_or_slug when inheriting from "OrganizationEndpoint"?

ajay-sentry replied:
Oh yeah I see it https://github.com/getsentry/sentry/blob/47387323e764e828dd2bd956787504158a877400/src/sentry/api/bases/organization.py#L628-L631

ajay-sentry reviewed (approved):
makes sense ✅

adrianviquez commented:
@michelletran-codecov , will these changes need to be reflected in new endpoints like [this one](https://github.com/getsentry/sentry/pull/95181) (and future ones?) I can help if needed w/ the changes if needed 👌

GabeVillalobos commented on src/sentry/codecov/base.py:
```diff
@@ -2,17 +2,33 @@
+                provider="github",
+                external_id=owner,
+                organization_id=organization.id,
+            )
```
Got it, that makes sense. What are we expecting the `owner` kwarg to be exactly? I see `external_id` listed on our integration-side as the github installation ID:
https://github.com/getsentry/sentry/blob/083bb2394b9dc7a7afc695d19e3c23e987de5cef/src/sentry/integrations/github/integration.py#L734
so I want to double check this query.

We should probably filter on integration status as well, since I don't think we'd want to allow access for anything disabled or pending deletion.

michelletran-sentry committed:
Update the Repository Tests to include integration validation

michelletran-sentry committed:
Use integration id to reference owner

This is to avoid needing to change the `get_integration` interface.
I've updated the endpoint to include the integration object in the request params,
so we can use that to get the owner slug.

ajay-sentry commented on src/sentry/codecov/base.py:
```diff
@@ -2,17 +2,42 @@
+        except (TypeError, ValueError):
+            raise ResourceDoesNotExist
+
+        if owner_id is not None and organization:
```
jw, how come we gotta do an is not None for owner_id but not for organization, or similarly have `if owner_id and organization`

I'm interpreting this as us covering for the owner_id is 0 case but I don't think that's an actual owner id

michelletran-codecov replied:
Hmm, I don't think we need to check for None now. And if the owner id is 0, then it wouldn't find a match in our query, so I think that should be fine. But thanks for the check! I'll remove the conditional.

michelletran-sentry committed:
Update branches endpoint with updated API Permissions
